infra/blueprint-test/pkg/tft/terraform.go

```diff
@@ -481,22 +488,50 @@ func (b *TFBlueprintTest) DefaultApply(assert *assert.Assertions) {
 
 // Init runs the default or custom init function for the blueprint.
 func (b *TFBlueprintTest) Init(assert *assert.Assertions) {
+	// allow only single write as Terraform plugin cache isn't concurrent safe
+	if err := b.tftCacheMutex.Lock(); err != nil {
+		b.t.Fatalf("Could not acquire lock: %v", err)
+	}
 	b.init(assert)
+	if err := b.tftCacheMutex.Unlock(); err != nil {
```

Comment:
It maybe good to wrap unlock in defer incase b.init panics/fails.

Reply:
Sure - as it returns `error` I can wrap in `func()`